src/convenience.jl

```diff
@@ -1,75 +1,85 @@
-
-iscam{T<:ImagineSignal}(com::T)  = iscam(daq_channel(com), rig_name(com))
-findcameras{T<:ImagineSignal}(coms::AbstractVector{T}) = find(x->iscam(x), coms)
-getcameras{T<:ImagineSignal}(coms::AbstractVector{T}) = view(coms, findcameras(coms))
-
-iscammonitor{T<:ImagineSignal}(com::T)  = iscammonitor(daq_channel(com), rig_name(com))
-findcameramonitors{T<:ImagineSignal}(coms::AbstractVector{T}) = find(x->iscammonitor(x), coms)
-getcameramonitors{T<:ImagineSignal}(coms::AbstractVector{T}) = view(coms, findcameramonitors(coms))
-
-islas{T<:ImagineSignal}(com::T)  = islas(daq_channel(com), rig_name(com))
-findlasers{T<:ImagineSignal}(coms::AbstractVector{T}) = find(x->islas(x), coms)
-getlasers{T<:ImagineSignal}(coms::AbstractVector{T}) = view(coms, findlasers(coms))
-
-isstim{T<:ImagineSignal}(com::T)  = isstim(daq_channel(com), rig_name(com))
-findstimuli{T<:ImagineSignal}(coms::AbstractVector{T}) = find(x->isstim(x), coms)
-getstimuli{T<:ImagineSignal}(coms::AbstractVector{T}) = view(coms, findstimuli(coms))
-
-isgalvo{T<:ImagineSignal}(com::T)  = iscam(daq_channel(com), rig_name(com))
-findgalvos{T<:ImagineSignal}(coms::AbstractVector{T}) = find(x->iscam(x), coms)
-getgalvos{T<:ImagineSignal}(coms::AbstractVector{T}) = view(coms, findcameras(coms))
-
-isgalvomonitor{T<:ImagineSignal}(com::T)  = isgalvomonitor(daq_channel(com), rig_name(com))
-findgalvomonitors{T<:ImagineSignal}(coms::AbstractVector{T}) = find(x->isgalvomonitor(x), coms)
-getgalvomonitors{T<:ImagineSignal}(coms::AbstractVector{T}) = view(coms, findgalvomonitors(coms))
-
-hasmonitor{T<:ImagineSignal}(com::T) = iscam(com) || ispos(com) || isgalvo(com)
-hasactuator{T<:ImagineSignal}(com::T) = iscammonitor(com) || isposmonitor(com) || isgalvomonitor(com)
-
-function monitor_name{T<:ImagineSignal}(com::T)
+finddaqchan(coms::AbstractVector{T}, nm::AbstractString) where{T<:ImagineSignal} =
+    findfirst(x->daq_channel(x) == nm, coms)
+getdaqchan(coms::AbstractVector{T}, nm::AbstractString) where{T<:ImagineSignal} =
+    coms[finddaqchan(coms, nm)]
+
+isdigital(com::T) where{T<:ImagineSignal} =
+    isdigital(daq_channel(com), rig_name(com))
+finddigital(coms::AbstractVector{T}) where{T<:ImagineSignal} =
+    findall(x->isdigital(x), coms)
+getdigital(coms::AbstractVector{T}) where{T<:ImagineSignal} =
+    view(coms, finddigital(coms))
+isanalog(daq_chan::AbstractString, rig::AbstractString) where{T<:ImagineSignal} =
+    !isdigital(daq_chan, rig)
+isanalog(com::T) where{T<:ImagineSignal} =
+    isanalog(daq_channel(com), rig_name(com))
+findanalog(coms::AbstractVector{T}) where{T<:ImagineSignal} =
+    findall(x->!isdigital(x), coms)
+getanalog(coms::AbstractVector{T}) where{T<:ImagineSignal} =
+    view(coms, findanalog(coms))
+
+isoutput(com::T) where{T<:ImagineSignal} = isoutput(daq_channel(com), rig_name(com))
+isinput(com::T) where{T<:ImagineSignal} = !isoutput(com)
+findoutputs(coms::AbstractVector{T}) where{T<:ImagineSignal} = findall(x->isoutput(x), coms)
```

Comment:
`findall(x->isoutput(x), coms)` can be more simply written as `findall(isoutput, coms)`. You're less likely to get weird `#3##4(x)` entries in your backtraces, too :smile: